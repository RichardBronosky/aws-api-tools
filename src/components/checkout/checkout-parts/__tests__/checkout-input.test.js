import React from 'react'
import renderer from 'react-test-renderer'
import CheckoutInput from '../checkout-input'

describe('CheckoutInput', () => {
  const testProps = {
    className: 'test',
    type: 'text',
    field: 'testField',
    label: 'Test Label',
    info: 'testInfo',
    setInfo: jest.fn(),
    afterComponent: null,
    name: 'Test Name',
    radioValue: null,
    invalidFields: [],
    required: false,
  }

  it('renders correctly as a standard text input', () => {
    const tree = renderer.create(<CheckoutInput { ...testProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly as a email type input', () => {
    const tree = renderer.create(<CheckoutInput { ...testProps } type="email" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly as a tel type input', () => {
    const tree = renderer.create(<CheckoutInput { ...testProps } type="tel" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly as a radio type input', () => {
    const tree = renderer.create(<CheckoutInput { ...testProps } type="radio" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly as a text type input with invalid data', () => {
    const tree = renderer.create(<CheckoutInput { ...testProps } invalidFields={ ['testField'] } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly as a radio type input an after component', () => {
    const tree = renderer.create(<CheckoutInput { ...testProps } afterComponent={ <p>after component test</p> } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly as a text type input that is required', () => {
    const tree = renderer.create(<CheckoutInput { ...testProps } required />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it('renders correctly while loading', () => {
  //   const newTestProps = {
  //     ...testProps,
  //     loading: true,
  //   }
  //   const tree = renderer.create(<CheckoutInput { ...newTestProps } />).toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
})
