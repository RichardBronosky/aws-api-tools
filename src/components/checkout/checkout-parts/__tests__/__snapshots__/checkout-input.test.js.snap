// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutInput renders correctly as a email type input 1`] = `
<label
  className="label testField"
  htmlFor="testField"
>
  Test Label
   
  <input
    aria-invalid={null}
    aria-required={false}
    checked={false}
    className="test"
    id="testField"
    name="Test Label"
    onChange={[Function]}
    onClick={[Function]}
    placeholder="Test Label"
    type="email"
  />
  <span />
</label>
`;

exports[`CheckoutInput renders correctly as a radio type input 1`] = `
Array [
  <input
    aria-invalid={null}
    aria-required={false}
    checked={true}
    className="test"
    id="testField"
    name="Test Name"
    onChange={[Function]}
    onClick={[Function]}
    placeholder=""
    type="radio"
    value={null}
  />,
  <span />,
]
`;

exports[`CheckoutInput renders correctly as a radio type input an after component 1`] = `
<label
  className="label testField"
  htmlFor="testField"
>
  Test Label
   
  <input
    aria-invalid={null}
    aria-required={false}
    checked={false}
    className="test"
    id="testField"
    name="Test Label"
    onChange={[Function]}
    onClick={[Function]}
    placeholder="Test Label"
    type="text"
  />
  <span>
    <p>
      after component test
    </p>
  </span>
</label>
`;

exports[`CheckoutInput renders correctly as a standard text input 1`] = `
<label
  className="label testField"
  htmlFor="testField"
>
  Test Label
   
  <input
    aria-invalid={null}
    aria-required={false}
    checked={false}
    className="test"
    id="testField"
    name="Test Label"
    onChange={[Function]}
    onClick={[Function]}
    placeholder="Test Label"
    type="text"
  />
  <span />
</label>
`;

exports[`CheckoutInput renders correctly as a tel type input 1`] = `
<label
  className="label testField"
  htmlFor="testField"
>
  Test Label
   
  <input
    aria-invalid={null}
    aria-required={false}
    checked={false}
    className="test"
    id="testField"
    name="Test Label"
    onChange={[Function]}
    onClick={[Function]}
    placeholder="Test Label"
    type="tel"
  />
  <span />
</label>
`;

exports[`CheckoutInput renders correctly as a text type input that is required 1`] = `
<label
  className="label testField"
  htmlFor="testField"
>
  Test Label*
   
  <input
    aria-invalid={true}
    aria-required={true}
    checked={false}
    className="test"
    id="testField"
    name="Test Label"
    onChange={[Function]}
    onClick={[Function]}
    placeholder="Test Label"
    type="text"
  />
  <span />
</label>
`;

exports[`CheckoutInput renders correctly as a text type input with invalid data 1`] = `
<label
  className="label testField"
  htmlFor="testField"
>
  Test Label
   
  <input
    aria-invalid={null}
    aria-required={false}
    checked={false}
    className="test invalid"
    id="testField"
    name="Test Label"
    onChange={[Function]}
    onClick={[Function]}
    placeholder="Test Label"
    type="text"
  />
  <span />
</label>
`;
